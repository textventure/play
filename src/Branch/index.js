import React, { Component } from 'react';
import Typography from '@material-ui/core/Typography';
import Card from '../Card';
import Choice from '../Choice';
import { getKey, getValue } from '../helpers/util';

export default class Branch extends Component {
  static defaultProps = {
    classes: {},
  };

  render() {
    const { classes, text, choices, selectChoice } = this.props;

    return (
      <Card>
        {text && (
          <Typography
            className={classes.cardContent}
            component="p"
            gutterBottom
          >
            {text}
          </Typography>
        )}

        {choices instanceof Array &&
          choices.map((choice, index) => (
            <Choice
              choiceId={getValue(choice)}
              className={classes.button}
              key={index}
              selectChoice={selectChoice}
            >
              {getKey(choice)}
            </Choice>
          ))}
      </Card>
    );
  }
}
