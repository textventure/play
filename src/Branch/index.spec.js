import React from 'react';
import { shallow } from 'enzyme';
import Branch from '.';

let wrapper;

describe('with props', () => {
  const props = {
    text: 'text',
    classes: {
      button: 'button',
      cardContent: 'cardContent',
    },
    choices: [{ key1: 'value1' }, { key2: 'value2' }],
    selectChoice: Function,
  };

  beforeAll(() => {
    wrapper = shallow(<Branch {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('passes props.selectChoice to <Choice>', () => {
    wrapper.find('Choice').forEach(choice => {
      expect(choice.prop('selectChoice')).toBe(props.selectChoice);
    });
  });
});
