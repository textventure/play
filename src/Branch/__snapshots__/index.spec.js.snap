// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with props renders correctly 1`] = `
<WithStyles(Card)>
  <WithStyles(Typography)
    className="cardContent"
    component="div"
    gutterBottom={true}
  >
    text
  </WithStyles(Typography)>
  <Choice
    choiceId="value1"
    className="button"
    selectChoice={[Function]}
  >
    <span>
      key1
    </span>
  </Choice>
  <Choice
    choiceId="value2"
    className="button"
    selectChoice={[Function]}
  >
    <span>
      key2
    </span>
  </Choice>
</WithStyles(Card)>
`;
